src/node_credentials.cc

```diff
@@ -87,14 +85,41 @@ bool SafeGetenv(const char* key,
 
   // Fallback to system environment which reads the real environment variable
   // through uv_os_getenv.
-  if (env_vars == nullptr) {
+  std::shared_ptr<KVStore> env_vars;
+  if (env == nullptr) {
     env_vars = per_process::system_environment;
+  } else {
+    env_vars = env->env_vars();
   }
 
   std::optional<std::string> value = env_vars->Get(key);
-  if (!value.has_value()) return false;
-  *text = value.value();
-  return true;
+
+  bool has_env = value.has_value();
+  if (has_env) {
+    *text = value.value();
+  }
+
+  auto options =
+      (env != nullptr ? env->options()
+                      : per_process::cli_options->per_isolate->per_env);
+
+  if (options->trace_env) {
+    fprintf(stderr, "[--trace-env] get environment variable \"%s\": ", key);
+    if (has_env) {
+      fprintf(stderr, "\"%s\"\n", text->c_str());
+    } else {
+      fprintf(stderr, "undefined\n");
+    }
+
+    if (options->trace_env_native_stack) {
```

Comment:
Can we use `PrintTraceEnvStack` here as well?

Reply:
Done, used the EnvironmentOptions as argument instead since this place can also be hit when an Environment is not yet created (nullptr here).